frontend/public/components/storage/shared.ts

```diff
@@ -1,55 +1,120 @@
+  [AccessModes.ROX, VolumeModes.Block],
+  ['All'],
+];
+
+type ProvisionerAccessModeMapping = {
+  [provisioner: string]: AllowedCombinations;
+};
+
+// See https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes for more details
+export const provisionerAccessModeMapping: ProvisionerAccessModeMapping = {
+  'kubernetes.io/no-provisioner': [allowedCombinations[0], allowedCombinations[1]],
+  'kubernetes.io/aws-ebs': [allowedCombinations[0], allowedCombinations[1]],
+  'kubernetes.io/gce-pd': [
+    allowedCombinations[0],
+    allowedCombinations[1],
+    allowedCombinations[4],
+    allowedCombinations[5],
+  ],
+  'kubernetes.io/glusterfs': [allowedCombinations[6]],
+  'kubernetes.io/cinder': [allowedCombinations[0], allowedCombinations[1]],
+  'kubernetes.io/azure-file': [allowedCombinations[6]],
+  'kubernetes.io/azure-disk': [allowedCombinations[0], allowedCombinations[1]],
+  'kubernetes.io/quobyte': [allowedCombinations[6]],
+  'kubernetes.io/rbd': [
+    allowedCombinations[0],
+    allowedCombinations[1],
+    allowedCombinations[4],
+    allowedCombinations[5],
+  ],
+  'kubernetes.io/vsphere-volume': [
+    allowedCombinations[0],
+    allowedCombinations[1],
+    allowedCombinations[2],
+    allowedCombinations[3],
+  ],
+  'kubernetes.io/portworx-volume': [
+    allowedCombinations[0],
+    allowedCombinations[1],
+    allowedCombinations[2],
+    allowedCombinations[3],
+  ],
+  'kubernetes.io/scaleio': [
+    allowedCombinations[0],
+    allowedCombinations[1],
+    allowedCombinations[4],
+    allowedCombinations[5],
+  ],
+  'kubernetes.io/storageos': [allowedCombinations[0], allowedCombinations[1]],
   // Since 4.6 new provisioners names will be without the 'kubernetes.io/' prefix.
-  'manila.csi.openstack.org': ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'],
-  'ebs.csi.aws.com': ['ReadWriteOnce'],
-  'csi.ovirt.org': ['ReadWriteOnce'],
-  'cinder.csi.openstack.org': ['ReadWriteOnce'],
-  'pd.csi.storage.gke.io': ['ReadWriteOnce'],
+  'manila.csi.openstack.org': [allowedCombinations[6]],
+  'ebs.csi.aws.com': [allowedCombinations[0], allowedCombinations[1]],
+  'csi.ovirt.org': [allowedCombinations[0], allowedCombinations[1]],
+  'cinder.csi.openstack.org': [allowedCombinations[0], allowedCombinations[1]],
+  'pd.csi.storage.gke.io': [allowedCombinations[0], allowedCombinations[1]],
+  'openshift-storage.cephfs.csi.ceph.com': [
+    allowedCombinations[0],
+    allowedCombinations[2],
+    allowedCombinations[4],
+  ],
+  'openshift-storage.rbd.csi.ceph.com': [
+    allowedCombinations[0],
+    allowedCombinations[4],
+    allowedCombinations[1],
+    allowedCombinations[3],
+    allowedCombinations[5],
+  ],
 };
-export const initialAccessModes = ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'];
+
+export const initialAccessModes: string[] = Object.values(AccessModes);
+export const initialVolumeModes: string[] = Object.values(VolumeModes);
+
```

Comment:
The reason I am still keeping string[] here and below function is, a lot of plugins are using this, if I want to return as enum i have to change their plugin also, that is a little bit difficult task for n

Reply:
> The reason I am still keeping string[] here and below function is, a lot of plugins are using this, if I want to return as enum i have to change their plugin also, that is a little bit difficult task for n

I might just remove the use of enum and keep it as a string for now. We could create a type for type checking like

```
export type AccessMode = 'ReadWriteOnce' | 'ReadWriteMany' | 'ReadOnlyMany';
```

Reply:
ack